xinference/core/worker.py

```diff
@@ -185,21 +185,33 @@ def _check_model_is_valid(self, model_name: str, model_format: Optional[str]):
     @log_sync(logger=logger)
     def register_model(self, model_type: str, model: str, persist: bool):
         # TODO: centralized model registrations
-        if model_type == "LLM":
+        if model_type == "LLM" or model_type == "embedding":
```

Comment:
Same as the comments above, if you don't want to write too many conditions, use a dict to handle different types, like {"model_type": [spec_cls, register_func]}.

Reply:
Fix